File: server/php/src/Application/Domain/Payments/PaymentDetails/PaymentProduct840SpecificOutput.php
<?php

namespace OnlinePayments\ExampleApp\Application\Domain\Payments\PaymentDetails;

final class PaymentProduct840SpecificOutput
{
    public ?Address $billingAddress = null;
    public ?PaymentProduct840CustomerAccount $customerAccount = null;
    public ?Address $customerAddress = null;
    public ?ProtectionEligibility $protectionEligibility = null;

    public function toArray(): array
    {
        return [
            'billingAddress' => ($this->billingAddress ?? new Address())->toArray(),
            'customerAccount' => ($this->customerAccount ?? new PaymentProduct840CustomerAccount())->toArray(),
            'customerAddress' => ($this->customerAddress ?? new Address())->toArray(),
            'protectionEligibility' => ($this->protectionEligibility ?? new ProtectionEligibility())->toArray(),
        ];
    }
}

File: server/dotnet/sdk-dotnet-example/Business/Domain/Payments/PaymentDetails/PaymentProduct840SpecificOutput.cs
namespace Business.Domain.Payments.PaymentDetails;

public class PaymentProduct840SpecificOutput
{
    public Address? BillingAddress { get; set; }

    public PaymentProduct840CustomerAccount? CustomerAccount { get; set; }

    public Address? CustomerAddress { get; set; }

    public ProtectionEligibility? ProtectionEligibility { get; set; }
}